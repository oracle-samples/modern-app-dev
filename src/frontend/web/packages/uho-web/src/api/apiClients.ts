/*
 * Copyright (c) 2023 Oracle and/or its affiliates.
 * Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracl.com/licenses/upl/
 */
export { AppointmentApi } from '@uho/appointment-api-client';
export { EncounterApi } from '@uho/encounter-api-client';
export { FrontendApi } from './frontendApi';
export { PatientApi } from '@uho/patient-api-client/dist/api-client';
export { ProviderApi } from '@uho/provider-api-client/dist/api-client';
