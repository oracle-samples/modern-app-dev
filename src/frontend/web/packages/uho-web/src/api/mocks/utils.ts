/*
 * Copyright (c) 2023 Oracle and/or its affiliates.
 * Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracl.com/licenses/upl/
 */

type Options = {
  delay?: number;
  error?: Error;
};
const defaultDelay = 0;

export function mock<T>(data: T, options?: Options): Promise<T> {
  return new Promise<T>((resolve, reject) => {
    setTimeout(() => {
      options?.error ? reject(options.error) : resolve(data);
    }, options?.delay || defaultDelay);
  });
}
export function reject<T>(options?: Options): Promise<T> {
  return new Promise<T>((_, reject) => {
    setTimeout(() => {
      reject(options?.error);
    }, options?.delay || defaultDelay);
  });
}

export function mockDate() {
  return new Date(1654733487565);
}
