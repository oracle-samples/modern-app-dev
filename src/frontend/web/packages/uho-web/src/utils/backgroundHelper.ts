/*
 * Copyright (c) 2023 Oracle and/or its affiliates.
 * Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracl.com/licenses/upl/
 */
import { ComponentProps } from 'preact/compat';
import { Avatar } from 'oj-c/avatar';

type AvatarProps = ComponentProps<typeof Avatar>;
const backgrounds: AvatarProps['background'][] = [
  'neutral',
  'orange',
  'green',
  'teal',
  'blue',
  'slate',
  'pink',
  'purple',
  'lilac',
  'gray'
];

// simple hash function to generate a number from 0 - brackets
function computeHash(key: string, brackets: number) {
  const hash = key.length + key.charCodeAt(0);
  return hash % brackets || 0;
}

export function getBackgroundByName(name: string) {
  return backgrounds[computeHash(name, backgrounds.length)];
}
