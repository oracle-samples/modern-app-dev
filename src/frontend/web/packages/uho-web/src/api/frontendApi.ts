/*
 * Copyright (c) 2023 Oracle and/or its affiliates.
 * Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracl.com/licenses/upl/
 */
import { Patient } from '@uho/patient-api-client/dist/api-client';
import { Provider } from '@uho/provider-api-client/dist/api-client';
import { UserRole } from '../utils/authProvider';

export type Apm = {
  serviceName: string;
  webApplication: string;
  ociDataUploadEndpoint: string;
  publicDataKey: string;
};

export type User = (Patient | Provider) & { role: UserRole };

export class Configuration {
  basePath: string;
  constructor(config: { basePath: string }) {
    this.basePath = config.basePath;
  }
}

export class FrontendApi {
  public constructor(public configuration: Configuration) {}

  public getUserInformation(): Promise<User> {
    return fetch(`${this.configuration.basePath}/userInformation`, {
      method: 'GET',
      headers: {
        accept: '*/*'
      }
    })
      .then((result) => result.json())
      .then((users) => users[0]);
  }

  public getApmInformation(): Promise<Apm> {
    return fetch(`${this.configuration.basePath}/apmInformation`, {
      method: 'GET',
      headers: {
        accept: '*/*'
      }
    }).then((result) => result.json());
  }

  public login(role: UserRole) {
    window.location.href = `oauth/login/${role.toLowerCase()}`;
  }

  public signout() {
    window.location.href = '/home/oauth/logout/';
  }
}
