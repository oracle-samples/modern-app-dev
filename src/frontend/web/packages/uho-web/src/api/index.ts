/*
 * Copyright (c) 2023 Oracle and/or its affiliates.
 * Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracl.com/licenses/upl/
 */
import { PatientApi, EncounterApi, AppointmentApi, ProviderApi, FrontendApi } from './apiClients';
import { Configuration as FrontendConfiguration } from './frontendApi';
import { Configuration as ProviderConfiguration } from '@uho/provider-api-client/dist/api-client';
import { Configuration as EncounterConfiguration, EncounterApiInterface } from '@uho/encounter-api-client';
import { Configuration as AppointmentConfiguration, AppointmentApiInterface } from '@uho/appointment-api-client';
import { Configuration as PatientConfiguration, PatientApiInterface } from '@uho/patient-api-client/dist/api-client';

const basePath = '/home/api';

export const providerApi = new ProviderApi(
  new ProviderConfiguration({
    basePath
  })
);
export const patientApi: PatientApiInterface = new PatientApi(new PatientConfiguration({ basePath }));
export const encounterApi: EncounterApiInterface = new EncounterApi(new EncounterConfiguration({ basePath }));
export const appointmentApi: AppointmentApiInterface = new AppointmentApi(new AppointmentConfiguration({ basePath }));
export const frontendApi: FrontendApi = new FrontendApi(new FrontendConfiguration({ basePath }));
