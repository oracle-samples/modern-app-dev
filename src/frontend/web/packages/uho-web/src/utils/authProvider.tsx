/*
 * Copyright (c) 2023 Oracle and/or its affiliates.
 * Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracl.com/licenses/upl/
 */
import { h, ComponentChildren, createContext } from 'preact';
import { useCallback, useContext } from 'preact/hooks';
import { RouteProps } from 'react-router';
import { Navigate } from 'react-router-dom';
import { Loading } from 'components/loading';
import { useLocalStorage } from 'hooks/useLocalStorage';
import { frontendApi } from 'api';
import { User } from 'api/frontendApi';
import { useQuery } from '@tanstack/react-query';
import { warn } from 'ojs/ojlogger';

export type UserRole = 'PATIENT' | 'PROVIDER';

type Context = {
  user: User | null;
  signin: () => Promise<User | null>;
  signout: () => void;
};

function useProvideAuth(): Context {
  const [user, setUser] = useLocalStorage<User | null>('user');

  const signin = useCallback(async () => {
    let userInfo: User | null = null;
    try {
      userInfo = await frontendApi.getUserInformation();
    } catch {
      warn('No active user session found.');
    }
    setUser(userInfo);
    return userInfo;
  }, []);

  const signout = useCallback(async () => {
    setUser(null);
    return frontendApi.signout();
  }, []);

  return {
    user,
    signin,
    signout
  };
}

// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
export const AuthContext = createContext<Context>(undefined!);

export function useAuth() {
  return useContext(AuthContext);
}

export function AuthProvider({ children }: { children: ComponentChildren }) {
  const auth = useProvideAuth();
  return <AuthContext.Provider value={auth}>{children}</AuthContext.Provider>;
}

export function RequireAuth({ children, role }: { children?: ComponentChildren; role: UserRole } & RouteProps) {
  const { signin } = useAuth();
  const { data, isLoading, isError } = useQuery<User | null, void>(['signin'], signin);

  if (isLoading) {
    return <Loading />;
  }
  if (!data || isError) {
    return <Navigate to="/" />;
  }
  if (data.role !== role) {
    return <Navigate to="/404" />;
  }
  return children;
}
