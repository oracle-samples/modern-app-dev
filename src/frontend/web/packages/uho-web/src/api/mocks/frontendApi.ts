/*
 * Copyright (c) 2023 Oracle and/or its affiliates.
 * Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracl.com/licenses/upl/
 */
import { UserRole } from '../../utils/authProvider';
import { Apm, Configuration, FrontendApi, User } from '../frontendApi';
import { mock } from './utils';

/**
 * Mocked Frontend API for local development
 */
export class MockFrontendApi extends FrontendApi {
  public constructor(configuration: Configuration) {
    super(configuration);
  }

  public getUserInformation(): Promise<User> {
    const userString = localStorage.getItem('user');
    if (userString) {
      return Promise.resolve(JSON.parse(userString));
    }
    return Promise.reject();
  }

  public getApmInformation(): Promise<Apm> {
    return mock({
      publicDataKey: 'publicDataKey',
      ociDataUploadEndpoint: 'ociDataUploadEndpoint',
      serviceName: 'serviceName',
      webApplication: 'webApplication'
    });
  }

  public async login(role: UserRole): Promise<void> {
    const user = {
      role,
      name: role === 'PATIENT' ? 'Patient XYZ' : 'Provider XYZ',
      username: role === 'PATIENT' ? 'DevUserId' : 'jsmith',
      email: role === 'PATIENT' ? 'patient@oracle.com' : 'provider@oracle.com'
    };
    localStorage.setItem('user', JSON.stringify(user));
    window.location.reload();
  }

  public signout() {
    window.location.href = '/home';
  }
}
