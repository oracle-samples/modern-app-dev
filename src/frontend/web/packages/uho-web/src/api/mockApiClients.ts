/*
 * Copyright (c) 2023 Oracle and/or its affiliates.
 * Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracl.com/licenses/upl/
 */
import { MockAppointmentApi as AppointmentApi } from './mocks/appointmentApi';
import { MockEncounterApi as EncounterApi } from './mocks/encounterApi';
import { MockFrontendApi as FrontendApi } from './mocks/frontendApi';
import { MockPatientApi as PatientApi } from './mocks/patientApi';
import { MockProviderApi as ProviderApi } from './mocks/providerApi';

export { AppointmentApi, EncounterApi, FrontendApi, PatientApi, ProviderApi };
