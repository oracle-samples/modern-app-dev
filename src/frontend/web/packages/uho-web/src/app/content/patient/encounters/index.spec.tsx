/*
 * Copyright (c) 2023 Oracle and/or its affiliates.
 * Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracl.com/licenses/upl/
 */
import { Encounters } from '.';
import { h } from 'preact';
import { queryClient, snapshot } from '../../../../../test/jetHelper';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../../../utils/authProvider';
import { QueryClientProvider } from '@tanstack/react-query';
import { User } from 'api/frontendApi';

const user: User = {
  role: 'PATIENT'
};

snapshot(
  'Patient: Encounters',
  <QueryClientProvider client={queryClient}>
    <AuthContext.Provider value={{ user, signin: () => Promise.resolve(user), signout: () => console.log('signout') }}>
      <MemoryRouter>
        <Encounters
        // location={{} as unknown as Props['location']}
        // history={{} as unknown as Props['history']}
        // match={{ params: { encounterId: '1' }, isExact: true, path: '/encounters', url: 'url' }}
        />
      </MemoryRouter>
    </AuthContext.Provider>
  </QueryClientProvider>
);
