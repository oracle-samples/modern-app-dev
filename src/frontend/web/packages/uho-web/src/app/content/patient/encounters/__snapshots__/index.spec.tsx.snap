// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests for Patient: Encounters Snapshot: Component - Patient: Encounters 1`] = `
<DocumentFragment>
  <div
    data-oj-binding-provider="preact"
  >
    <div
      class="overview-layout oj-sm-padding-12x-horizontal oj-flex oj-sm-flex-direction-column fullscreen"
    >
      <div
        class="oj-flex-item oj-sm-padding-6x-top oj-sm-padding-4x-bottom oj-sm-flex-0"
      >
        <div
          class="oj-flex-bar"
        >
          <div
            class="oj-flex-bar-start oj-flex oj-sm-flex-direction-column"
          >
            <div
              class="oj-typography-subheading-lg oj-flex-item header-title-text-color"
            >
              Encounters
            </div>
            <div
              class="oj-typography-semi-bold-sm oj-flex-item header-subtitle-text-color"
            >
              See all encounters
            </div>
          </div>
          <div
            class="oj-flex-bar-end"
          />
        </div>
      </div>
      <div
        class="oj-flex-item oj-sm-flex-0"
      >
        <div
          class="banner"
        />
      </div>
      <div
        class="oj-flex-item oj-flex oj-sm-flex-1"
      >
        <div
          class="oj-flex-item oj-sm-only-hide secondary oj-sm-3 oj-sm-padding-8x"
        >
          <oj-list-view
            class="oj-listview oj-listview-container oj-component oj-listview-gridlines-hidden oj-complete gridline-top-hidden gridline-bottom-hidden"
          >
            <div
              class="oj-helper-detect-contraction"
            >
              <div
                style="width: 200%; height: 200%;"
              />
            </div>
            <div
              class="oj-helper-detect-expansion"
            >
              <div />
            </div>
            <div
              class="oj-helper-hidden-accessible"
              tabindex="-1"
            >
              Â 
            </div>
            <ul
              aria-busy="false"
              aria-label="list of encounters"
              aria-multiselectable="false"
              aria-rowcount="7"
              class="oj-listview-element oj-component-initnode"
              data-oj-context=""
              data-oj-internal=""
              role="grid"
              tabindex="0"
              x-ms-format-detection="none"
            >
              <li
                aria-rowindex="1"
                class="oj-listview-focused-element oj-listview-item-element oj-listview-item oj-listview-item-layout"
                role="row"
              >
                <div
                  aria-selected="false"
                  class="oj-listview-cell-element"
                  role="gridcell"
                >
                  <oj-list-item-layout
                    class="oj-complete"
                    data-oj-vdom-template-root=""
                  >
                    <div
                      class="oj-listitemlayout-grid"
                    >
                      <div
                        class="oj-listitemlayout-textslots"
                      >
                        <div
                          data-oj-tabmod="0"
                          tabindex="-1"
                        >
                          <div
                            class="oj-typography-body-md"
                          >
                            Choroidal hemorrhage
                          </div>
                        </div>
                        <div
                          data-oj-tabmod="0"
                          tabindex="-1"
                        >
                          <div
                            class="oj-typography-body-sm"
                            slot="secondary"
                          >
                            162673000
                          </div>
                        </div>
                        <div
                          class="oj-listitemlayout-tertiary"
                          data-oj-tabmod="0"
                          tabindex="-1"
                        >
                          <div
                            class="oj-typography-body-xs oj-text-color-secondary"
                            slot="tertiary"
                          >
                            Dr. Foo
                          </div>
                        </div>
                      </div>
                      <div
                        class="oj-listitemlayout-extra"
                      >
                        <div
                          class="oj-listitemlayout-action oj-listitemlayout-start-padding"
                          data-oj-clickthrough="disabled"
                        >
                          <div
                            class=""
                            slot="action"
                          />
                        </div>
                      </div>
                    </div>
                  </oj-list-item-layout>
                </div>
              </li>
              <li
                aria-rowindex="2"
                class="oj-listview-focused-element oj-listview-item-element oj-listview-item oj-listview-item-layout"
                role="row"
              >
                <div
                  aria-selected="false"
                  class="oj-listview-cell-element"
                  role="gridcell"
                >
                  <oj-list-item-layout
                    class="oj-complete"
                    data-oj-vdom-template-root=""
                  >
                    <div
                      class="oj-listitemlayout-grid"
                    >
                      <div
                        class="oj-listitemlayout-textslots"
                      >
                        <div
                          data-oj-tabmod="0"
                          tabindex="-1"
                        >
                          <div
                            class="oj-typography-body-md"
                          >
                            Choroidal hemorrhage
                          </div>
                        </div>
                        <div
                          data-oj-tabmod="0"
                          tabindex="-1"
                        >
                          <div
                            class="oj-typography-body-sm"
                            slot="secondary"
                          >
                            162673000
                          </div>
                        </div>
                        <div
                          class="oj-listitemlayout-tertiary"
                          data-oj-tabmod="0"
                          tabindex="-1"
                        >
                          <div
                            class="oj-typography-body-xs oj-text-color-secondary"
                            slot="tertiary"
                          >
                            Dr. Bar
                          </div>
                        </div>
                      </div>
                      <div
                        class="oj-listitemlayout-extra"
                      >
                        <div
                          class="oj-listitemlayout-action oj-listitemlayout-start-padding"
                          data-oj-clickthrough="disabled"
                        >
                          <div
                            class=""
                            slot="action"
                          />
                        </div>
                      </div>
                    </div>
                  </oj-list-item-layout>
                </div>
              </li>
              <li
                aria-rowindex="3"
                class="oj-listview-focused-element oj-listview-item-element oj-listview-item oj-listview-item-layout"
                role="row"
              >
                <div
                  aria-selected="false"
                  class="oj-listview-cell-element"
                  role="gridcell"
                >
                  <oj-list-item-layout
                    class="oj-complete"
                    data-oj-vdom-template-root=""
                  >
                    <div
                      class="oj-listitemlayout-grid"
                    >
                      <div
                        class="oj-listitemlayout-textslots"
                      >
                        <div
                          data-oj-tabmod="0"
                          tabindex="-1"
                        >
                          <div
                            class="oj-typography-body-md"
                          >
                            Choroidal hemorrhage
                          </div>
                        </div>
                        <div
                          data-oj-tabmod="0"
                          tabindex="-1"
                        >
                          <div
                            class="oj-typography-body-sm"
                            slot="secondary"
                          >
                            162673000
                          </div>
                        </div>
                        <div
                          class="oj-listitemlayout-tertiary"
                          data-oj-tabmod="0"
                          tabindex="-1"
                        >
                          <div
                            class="oj-typography-body-xs oj-text-color-secondary"
                            slot="tertiary"
                          >
                            Dr. Jonah
                          </div>
                        </div>
                      </div>
                      <div
                        class="oj-listitemlayout-extra"
                      >
                        <div
                          class="oj-listitemlayout-action oj-listitemlayout-start-padding"
                          data-oj-clickthrough="disabled"
                        >
                          <div
                            class=""
                            slot="action"
                          />
                        </div>
                      </div>
                    </div>
                  </oj-list-item-layout>
                </div>
              </li>
              <li
                aria-rowindex="4"
                class="oj-listview-focused-element oj-listview-item-element oj-listview-item oj-listview-item-layout"
                role="row"
              >
                <div
                  aria-selected="false"
                  class="oj-listview-cell-element"
                  role="gridcell"
                >
                  <oj-list-item-layout
                    class="oj-complete"
                    data-oj-vdom-template-root=""
                  >
                    <div
                      class="oj-listitemlayout-grid"
                    >
                      <div
                        class="oj-listitemlayout-textslots"
                      >
                        <div
                          data-oj-tabmod="0"
                          tabindex="-1"
                        >
                          <div
                            class="oj-typography-body-md"
                          >
                            Choroidal hemorrhage
                          </div>
                        </div>
                        <div
                          data-oj-tabmod="0"
                          tabindex="-1"
                        >
                          <div
                            class="oj-typography-body-sm"
                            slot="secondary"
                          >
                            162673000
                          </div>
                        </div>
                        <div
                          class="oj-listitemlayout-tertiary"
                          data-oj-tabmod="0"
                          tabindex="-1"
                        >
                          <div
                            class="oj-typography-body-xs oj-text-color-secondary"
                            slot="tertiary"
                          >
                            Dr. Zeus
                          </div>
                        </div>
                      </div>
                      <div
                        class="oj-listitemlayout-extra"
                      >
                        <div
                          class="oj-listitemlayout-action oj-listitemlayout-start-padding"
                          data-oj-clickthrough="disabled"
                        >
                          <div
                            class=""
                            slot="action"
                          />
                        </div>
                      </div>
                    </div>
                  </oj-list-item-layout>
                </div>
              </li>
              <li
                aria-rowindex="5"
                class="oj-listview-focused-element oj-listview-item-element oj-listview-item oj-listview-item-layout"
                role="row"
              >
                <div
                  aria-selected="false"
                  class="oj-listview-cell-element"
                  role="gridcell"
                >
                  <oj-list-item-layout
                    class="oj-complete"
                    data-oj-vdom-template-root=""
                  >
                    <div
                      class="oj-listitemlayout-grid"
                    >
                      <div
                        class="oj-listitemlayout-textslots"
                      >
                        <div
                          data-oj-tabmod="0"
                          tabindex="-1"
                        >
                          <div
                            class="oj-typography-body-md"
                          >
                            Choroidal hemorrhage
                          </div>
                        </div>
                        <div
                          data-oj-tabmod="0"
                          tabindex="-1"
                        >
                          <div
                            class="oj-typography-body-sm"
                            slot="secondary"
                          >
                            162673000
                          </div>
                        </div>
                        <div
                          class="oj-listitemlayout-tertiary"
                          data-oj-tabmod="0"
                          tabindex="-1"
                        >
                          <div
                            class="oj-typography-body-xs oj-text-color-secondary"
                            slot="tertiary"
                          >
                            Dr. Bar
                          </div>
                        </div>
                      </div>
                      <div
                        class="oj-listitemlayout-extra"
                      >
                        <div
                          class="oj-listitemlayout-action oj-listitemlayout-start-padding"
                          data-oj-clickthrough="disabled"
                        >
                          <div
                            class=""
                            slot="action"
                          />
                        </div>
                      </div>
                    </div>
                  </oj-list-item-layout>
                </div>
              </li>
              <li
                aria-rowindex="6"
                class="oj-listview-focused-element oj-listview-item-element oj-listview-item oj-listview-item-layout"
                role="row"
              >
                <div
                  aria-selected="false"
                  class="oj-listview-cell-element"
                  role="gridcell"
                >
                  <oj-list-item-layout
                    class="oj-complete"
                    data-oj-vdom-template-root=""
                  >
                    <div
                      class="oj-listitemlayout-grid"
                    >
                      <div
                        class="oj-listitemlayout-textslots"
                      >
                        <div
                          data-oj-tabmod="0"
                          tabindex="-1"
                        >
                          <div
                            class="oj-typography-body-md"
                          >
                            Choroidal hemorrhage
                          </div>
                        </div>
                        <div
                          data-oj-tabmod="0"
                          tabindex="-1"
                        >
                          <div
                            class="oj-typography-body-sm"
                            slot="secondary"
                          >
                            162673000
                          </div>
                        </div>
                        <div
                          class="oj-listitemlayout-tertiary"
                          data-oj-tabmod="0"
                          tabindex="-1"
                        >
                          <div
                            class="oj-typography-body-xs oj-text-color-secondary"
                            slot="tertiary"
                          >
                            Dr. Jonah
                          </div>
                        </div>
                      </div>
                      <div
                        class="oj-listitemlayout-extra"
                      >
                        <div
                          class="oj-listitemlayout-action oj-listitemlayout-start-padding"
                          data-oj-clickthrough="disabled"
                        >
                          <div
                            class=""
                            slot="action"
                          />
                        </div>
                      </div>
                    </div>
                  </oj-list-item-layout>
                </div>
              </li>
              <li
                aria-rowindex="7"
                class="oj-listview-focused-element oj-listview-item-element oj-listview-item oj-listview-item-layout"
                role="row"
              >
                <div
                  aria-selected="false"
                  class="oj-listview-cell-element"
                  role="gridcell"
                >
                  <oj-list-item-layout
                    class="oj-complete"
                    data-oj-vdom-template-root=""
                  >
                    <div
                      class="oj-listitemlayout-grid"
                    >
                      <div
                        class="oj-listitemlayout-textslots"
                      >
                        <div
                          data-oj-tabmod="0"
                          tabindex="-1"
                        >
                          <div
                            class="oj-typography-body-md"
                          >
                            Choroidal hemorrhage
                          </div>
                        </div>
                        <div
                          data-oj-tabmod="0"
                          tabindex="-1"
                        >
                          <div
                            class="oj-typography-body-sm"
                            slot="secondary"
                          >
                            162673000
                          </div>
                        </div>
                        <div
                          class="oj-listitemlayout-tertiary"
                          data-oj-tabmod="0"
                          tabindex="-1"
                        >
                          <div
                            class="oj-typography-body-xs oj-text-color-secondary"
                            slot="tertiary"
                          >
                            Dr. Zeus
                          </div>
                        </div>
                      </div>
                      <div
                        class="oj-listitemlayout-extra"
                      >
                        <div
                          class="oj-listitemlayout-action oj-listitemlayout-start-padding"
                          data-oj-clickthrough="disabled"
                        >
                          <div
                            class=""
                            slot="action"
                          />
                        </div>
                      </div>
                    </div>
                  </oj-list-item-layout>
                </div>
              </li>
              <li
                class="oj-listview-skeleton-container"
                role="presentation"
                style="visibility: hidden;"
              >
                <ul
                  class="oj-listview-group"
                  role="presentation"
                >
                  <li
                    class="oj-listview-item oj-listview-item-layout gridline-hidden"
                    role="presentation"
                  >
                    <div
                      class="oj-listview-cell-element oj-listview-skeleton oj-listview-skeleton-line-height oj-animation-skeleton"
                    />
                  </li>
                  <li
                    class="oj-listview-item oj-listview-item-layout gridline-hidden"
                    role="presentation"
                  >
                    <div
                      class="oj-listview-cell-element oj-listview-skeleton oj-listview-skeleton-line-height oj-animation-skeleton"
                    />
                  </li>
                  <li
                    class="oj-listview-item oj-listview-item-layout gridline-hidden"
                    role="presentation"
                  >
                    <div
                      class="oj-listview-cell-element oj-listview-skeleton oj-listview-skeleton-line-height oj-animation-skeleton"
                    />
                  </li>
                </ul>
              </li>
            </ul>
            <template
              slot="itemTemplate"
            />
            <div
              class="oj-listview-status-message oj-listview-status"
              role="status"
              style="display: none;"
            >
              <div
                class="oj-icon oj-listview-loading-icon"
              />
            </div>
            <div
              class="oj-helper-hidden-accessible"
              role="status"
              tabindex="-1"
            />
            <div
              class="oj-helper-hidden-accessible"
            >
              Use arrow keys to expand and collapse.
            </div>
          </oj-list-view>
        </div>
        <div
          class="oj-flex-item primary oj-sm-12 oj-md-9 oj-lg-9 oj-sm-padding-8x"
        >
          <div
            class="oj-flex oj-sm-flex-direction-column oj-sm-justify-content-flex-end empty-screen"
          >
            <span
              class="oj-text-primary-color title oj-typography-heading-2xl oj-flex-item oj-sm-flex-0"
            >
              No encounter selected
            </span>
            <span
              class="oj-text-primary-color description oj-typography-body-xl oj-flex-item oj-sm-flex-0"
            >
              Select an encounter or submit a new encounter
            </span>
          </div>
        </div>
      </div>
      <div
        class="oj-flex-item oj-sm-flex-0"
      >
        <div
          class="footer"
        >
          <oj-navigation-list
            class="oj-sm-condense oj-sm-margin-8x-horizontal oj-navigationlist oj-navigationlist-horizontal-bottom oj-navigationlist-expanded oj-navigationlist-horizontal oj-navigationlist-app-level oj-component-initnode oj-complete"
            role="tablist"
            tabindex="0"
          >
            <template
              slot="itemTemplate"
            />
            <div
              class="oj-navigationlist-listview-container"
              role="presentation"
            >
              <div
                class="oj-navigationlist-listview oj-component"
                role="presentation"
                style="max-width: none;"
              >
                <div
                  class="oj-helper-detect-contraction"
                >
                  <div
                    style="width: 200%; height: 200%;"
                  />
                </div>
                <div
                  class="oj-helper-detect-expansion"
                >
                  <div />
                </div>
                <div
                  class="oj-helper-hidden-accessible"
                  tabindex="-1"
                >
                  Â 
                </div>
                <ul
                  class="oj-navigationlist-element oj-navigationlist-has-icons"
                  role="presentation"
                >
                  <li
                    class="oj-navigationlist-item-element oj-navigationlist-item oj-default"
                    data-oj-vdom-template-root=""
                    role="presentation"
                  >
                    <a
                      aria-selected="false"
                      class="oj-navigationlist-focused-element oj-navigationlist-item-content"
                      data-oj-tabmod="0"
                      href="#"
                      role="tab"
                      tabindex="-1"
                    >
                      <span
                        class="oj-navigationlist-item-icon oj-ux-ico-home"
                      />
                      <span
                        class="oj-navigationlist-item-label"
                      >
                        Home
                      </span>
                    </a>
                  </li>
                  <li
                    class="oj-navigationlist-item-element oj-navigationlist-item oj-default"
                    data-oj-vdom-template-root=""
                    role="presentation"
                  >
                    <a
                      aria-selected="false"
                      class="oj-navigationlist-focused-element oj-navigationlist-item-content"
                      data-oj-tabmod="0"
                      href="#"
                      role="tab"
                      tabindex="-1"
                    >
                      <span
                        class="oj-navigationlist-item-icon oj-ux-ico-doctor"
                      />
                      <span
                        class="oj-navigationlist-item-label"
                      >
                        Providers
                      </span>
                    </a>
                  </li>
                  <li
                    class="oj-navigationlist-item-element oj-navigationlist-item oj-default"
                    data-oj-vdom-template-root=""
                    role="presentation"
                  >
                    <a
                      aria-selected="false"
                      class="oj-navigationlist-focused-element oj-navigationlist-item-content"
                      data-oj-tabmod="0"
                      href="#"
                      role="tab"
                      tabindex="-1"
                    >
                      <span
                        class="oj-navigationlist-item-icon oj-ux-ico-calendar-clock"
                      />
                      <span
                        class="oj-navigationlist-item-label"
                      >
                        Appointments
                      </span>
                    </a>
                  </li>
                  <li
                    class="oj-navigationlist-item-element oj-navigationlist-item oj-default"
                    data-oj-vdom-template-root=""
                    role="presentation"
                  >
                    <a
                      aria-selected="false"
                      class="oj-navigationlist-focused-element oj-navigationlist-item-content"
                      data-oj-tabmod="0"
                      href="#"
                      role="tab"
                      tabindex="-1"
                    >
                      <span
                        class="oj-navigationlist-item-icon oj-ux-ico-clock-history"
                      />
                      <span
                        class="oj-navigationlist-item-label"
                      >
                        Encounters
                      </span>
                    </a>
                  </li>
                </ul>
                <div
                  class="oj-navigationlist-status-message oj-navigationlist-status"
                  role="status"
                  style="display: none;"
                >
                  <div
                    class="oj-icon oj-navigationlist-loading-icon"
                  />
                </div>
                <div
                  class="oj-helper-hidden-accessible"
                  role="status"
                  tabindex="-1"
                />
              </div>
            </div>
          </oj-navigation-list>
        </div>
      </div>
    </div>
  </div>
  <div
    id="__oj_layerhost_container"
  />
</DocumentFragment>
`;
